pyabc/epsilon/temperature.py

```diff
@@ -0,0 +1,502 @@
+
+        # temperature is inverse beta
+        temperature = 1.0 / beta_opt
+        return temperature
+
+
+class ExponentialDecayScheme(TemperatureScheme):
+    """
+    If `max_nr_populations` is finite:
+
+    .. math::
+        T_j = T_{max}^{(n-j)/n}
+
+    where n denotes the number of populations, and j=1,...,n the iteration.
+    This translates to
+
+    .. math::
+        T_j = T_{j-1}^{(n-j)/(n-(j-1))}.
+
+    This ensures that a temperature of 1.0 is reached after exactly the
+    remaining number of steps.
+
+    If `max_nr_populations` is infinite, the next temperature is
+
+    .. math::
+        T_j = \\alpha \\cdot T_{j-1},
+
+    where by default alpha=0.25.
+
+    So, in both cases the sequence of temperatures follows an exponential
+    decay, also known as a geometric progression, or a linear progression
+    in log-space.
+
+    Note that the formula is applied anew in each iteration. That has the
+    advantage that, if also other schemes are used s.t. T_{j-1} is smaller
+    than by the above, advantage can be made of this.
+
+    Parameters
+    ----------
+
+    alpha: float
+        Factor by which to reduce the temperature, if `max_nr_populations`
+        is infinite.
+    """
+
+    def __init__(self, alpha: float = 0.25):
+        self.alpha = alpha
+
+    def __call__(self,
+                 t: int,
+                 get_weighted_distances: Callable,
+                 max_nr_populations: int,
+                 pdf_norm: float,
+                 kernel_scale: str,
+                 prev_temperature: float,
+                 acceptance_rate: float):
+        # needs a starting temperature
+        # if not available, return infinite temperature
+        if prev_temperature is None:
+            return np.inf
+
+        # base temperature
+        temp_base = prev_temperature
+
+        if max_nr_populations == np.inf:
+            # just decrease by a factor of alpha each round
+            temperature = self.alpha * temp_base
+            return temperature
+
+        # how many steps left?
+        t_to_go = max_nr_populations - t
+
+        # compute next temperature according to exponential decay
+        temperature = temp_base ** ((t_to_go - 1) / t_to_go)
+
+        return temperature
+
+
+class PolynomialDecayScheme(TemperatureScheme):
+    """
+    Compute next temperature as pre-last entry in
+
+    >>> np.linspace(1, (temp_base)**(1 / temp_decay_exponent),
+    >>>             t_to_go + 1) ** temp_decay_exponent)
+
+    Requires finite `max_nr_populations`.
+
+    Note that this is similar to the `scheme_exponential_decay`, which is
+    indeed the limit for `temp_decay_exponent -> infinity`. For smaller
+    exponent, the sequence makes larger steps for low temperatures. This
+    can be useful in cases, where lower temperatures (which are usually
+    more expensive) can be traversed in few larger steps.
+    """
+
+    def __init__(self, exponent: float = 3):
```

Comment:
why 3?

Reply:
good question. The value was found heuristically. However, also other exponents work. I will add a note on this.